### PR TITLE
docs(lua): correct generated `fname` param annotations

### PR DESCRIPTION
**Pull request by mrcjkb**
#24473 has a typo in the `fnamemodify` type annotation.

### CODE REVIEW

**Comment by mfussenegger** [comment]
This file is generated, see https://github.com/neovim/neovim/blob/1ee905a63aa9f3d6873ec75a0954866d8660699e/src/nvim/eval.lua#L2802 for the source

**Comment by seandewar** [comment]
Also seems other `fname` params in `eval.lua` are `integer` too; these should all be `string`.

**Comment by mrcjkb** [comment]
Looks like I can't run `gen_vimdoc.py` locally:
```console
Running ./scripts/gen_eval_files.lua
E5113: Error while calling lua chunk: ./scripts/gen_eval_files.lua:295: module 'src/nvim/eval' not found:
        no field package.preload['src/nvim/eval']
        no file '/nix/store/p41v49dj099qryslcd3wpm9qmgwnf3by-luajit-2.1.0-2022-10-04-env/share/lua/5.1/src/nvim/eval.lua'
        no file '/nix/store/p41v49dj099qryslcd3wpm9qmgwnf3by-luajit-2.1.0-2022-10-04-env/share/lua/5.1/src/nvim/eval/init.lua'
        no file '/nix/store/p41v49dj099qryslcd3wpm9qmgwnf3by-luajit-2.1.0-2022-10-04-env/lib/lua/5.1/src/nvim/eval.so'
```

**Comment by mrcjkb** [comment]
> Update: I'll see if I can run it in the nix devShell...

Nope, I'm getting the same error in the devShell.

**Comment by seandewar** [comment]
Probably doesn't have "./?.lua" in the `package.path` due to being installed through Nix.

Try running the script manually from the repo root like:
`nvim --cmd "lua package.path='./?.lua' -l scripts/gen_eval_files.lua`

**Comment by mrcjkb** [comment]
> Probably doesn't have "./?.lua" in the `package.path` due to being installed through Nix.
> 
> Try running the script manually from the repo root like: `nvim --cmd "lua package.path='./?.lua'" -l scripts/gen_eval_files.lua`

Thanks. After `--cmd "lua package.path='./?.lua'"` to the script's shebang and calling gen_vimdoc.py, it worked.

File: scripts/gen_eval_files.lua [hunk 1]
```diff
@@ -261,7 +261,7 @@ local function get_api_keysets()
   local ret = {} --- @type table<string, vim.EvalFn>
 
   --- @type {[1]: string, [2]: {[1]: string, [2]: string}[] }[]
-  local keysets = metadata.keysets
+  local keysets = metadata.keysets or {}
```
Comment (seandewar) [inline scripts/gen_eval_files.lua:264]: Why was this changed? The field should exist.

If this was to avoid building Nvim first, this isn't ideal as it'll wipe the `_meta/api_keysets.lua` file.

**Review by seandewar — APPROVED** [review]
LGTM, thanks!

**Review by justinmk — APPROVED** [review]
